test/testutil/fs.go

```diff
@@ -0,0 +1,62 @@
+package testutil
+
+import (
+	"embed"
+	"io/fs"
+	"strings"
+	"testing/fstest"
+)
+
+//go:embed testdata/*
+var testDataFS embed.FS
+
+// NewRegistryV1FS returns a new test filesystem that contains a valid registry+v1 bundle.
+func NewRegistryV1FS() fstest.MapFS {
+	fsys, err := makeMapFS("testdata/registry-v1/")
+	if err != nil {
+		panic(err)
+	}
+
+	return fsys
+}
+
+// NewPlainV0FS returns a new test filesystem that contains a valid plain+v0 bundle.
+func NewPlainV0FS() fstest.MapFS {
+	fsys, err := makeMapFS("testdata/plain-v0/")
+	if err != nil {
+		panic(err)
+	}
+
+	return fsys
+}
+
+// embed.FS is only walkable from the root directory so this function accepts a prefix to filter out unwanted files.
+// This looks like it could be a go bug if its not explicitly called out.
```

Comment:
Can you use https://pkg.go.dev/io/fs#Sub here? It returns an fs that is a subdirectory of the root FS.